search/api_test.py

```diff
@@ -1,365 +1,122 @@
-
-        assert serialize_program_enrolled_user(program_enrollment) == {
-            '_id': program_enrollment.id,
-            'id': program_enrollment.id,
-            'user_id': profile.user.id,
-            'profile': ProfileSerializer().to_representation(profile),
-            'program': UserProgramSerializer.serialize(program_enrollment)
-        }
-
-
-class GetConnTests(ESTestCase):
-    """
-    Tests for get_conn
-    """
-    def setUp(self):
-        """
-        Start without any index
-        """
-        super(GetConnTests, self).setUp()
-
-        conn = get_conn(verify=False)
-        index_name = settings.ELASTICSEARCH_INDEX
-        conn.indices.delete(index_name)
-
-        # Clear globals
-        from search import api
-        api._CONN = None  # pylint: disable=protected-access
-        api._CONN_VERIFIED = False  # pylint: disable=protected-access
-
-    def test_no_index(self):
-        """
-        Test that an error is raised if we don't have an index
-        """
-        with self.assertRaises(ReindexException) as ex:
-            get_conn()
-        assert str(ex.exception) == "Unable to find index {}".format(settings.ELASTICSEARCH_INDEX)
-
-    def test_no_mapping(self):
-        """
-        Test that error is raised if we don't have a mapping
-        """
-        conn = get_conn(verify=False)
-        conn.indices.create(settings.ELASTICSEARCH_INDEX)
-
-        with self.assertRaises(ReindexException) as ex:
-            get_conn()
-        assert str(ex.exception) == "Mapping {} not found".format(USER_DOC_TYPE)
-
-
-class RecreateIndexTests(ESTestCase):
-    """
-    Tests for management commands
-    """
-    def setUp(self):
-        """
-        Start without any index
-        """
-        super(RecreateIndexTests, self).setUp()
-        conn = get_conn(verify=False)
-        index_name = settings.ELASTICSEARCH_INDEX
-        if conn.indices.exists(index_name):
-            conn.indices.delete(index_name)
+        with self.assertRaises(NoProgramAccessException):
+            create_search_obj(profile.user)
 
-    def test_create_index(self):  # pylint: disable=no-self-use
+    def test_prepare_and_execute_search(self):
         """
-        Test that recreate_index will create an index and let search successfully
+        Test that a Search object is properly prepared and executed
         """
-        recreate_index()
-        assert es.search()['total'] == 0
+        mock_search_func = Mock(name='execute', return_value=['result1', 'result2'])
+        with patch('search.api.create_search_obj', autospec=True, return_value=None) as mock_create_search_obj:
+            results = prepare_and_execute_search(self.user, search_func=mock_search_func)
+            assert mock_create_search_obj.called
+            assert results == ['result1', 'result2']
 
-    def test_update_index(self):  # pylint: disable=no-self-use
+    def test_all_query_matching_emails(self):  # pylint: disable=no-self-use
```

Comment:
not used yet, no. i initially went about this trying to solve the general problem of taking a search query and paging through all of its results. i adapted that to the problem of paging through results and just retrieving the unique emails. having the 'base' class (`ESFullResultSearch`) is going to be beneficial in future issues for micromasters email. i'm happy to add a test case or two to assert that its working correctly